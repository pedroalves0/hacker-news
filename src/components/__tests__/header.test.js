import React from 'react';
import { shallow } from 'enzyme';
import Header from '../header/header';

describe('components/header', () => {
    test('should render correctly', () => {
        const wrapper = shallow(
            <Header isOffline={false}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should render a text info if the app is offline', () => {
        const wrapper = shallow(
            <Header isOffline={true}/>
        );

        expect(wrapper).toMatchSnapshot();
    });
});