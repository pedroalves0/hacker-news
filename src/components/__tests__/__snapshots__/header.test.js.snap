// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/header should render a text info if the app is offline 1`] = `
<div
  className="header"
>
  <span
    className="header-title"
  >
    <span
      className="t-boldi"
    >
      HACKER
    </span>
    <span
      className="t-bold"
    >
      NEWS
    </span>
  </span>
  <span
    className="header-offline"
  >
    <FaExclamationTriangle
      className="header-offline-icon"
    />
    <span
      className="header-offline-text"
    >
      App offline. No cached news found. Please refresh to try again.
    </span>
  </span>
</div>
`;

exports[`components/header should render correctly 1`] = `
<div
  className="header"
>
  <span
    className="header-title"
  >
    <span
      className="t-boldi"
    >
      HACKER
    </span>
    <span
      className="t-bold"
    >
      NEWS
    </span>
  </span>
</div>
`;
